libs/exec/exec.go

```diff
@@ -0,0 +1,249 @@
+package exec
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	osexec "os/exec"
+)
+
+type Command interface {
+	// Wait for command to terminate. It must have been previously started.
+	Wait() error
+
+	// StdinPipe returns a pipe that will be connected to the command's standard input when the command starts.
+	Stdout() io.ReadCloser
+
+	// StderrPipe returns a pipe that will be connected to the command's standard error when the command starts.
+	Stderr() io.ReadCloser
+}
+
+type command struct {
+	done   chan bool
+	err    chan error
+	stdout io.ReadCloser
+	stderr io.ReadCloser
+}
+
+func (c *command) Wait() error {
+	select {
+	case <-c.done:
+		return nil
+	case err := <-c.err:
+		return err
+	}
```

Comment:
This can be collapsed into a single channel. The current approach presents a race condition if both channels are closed: the select may proceed on either of them. If there was in fact an error, it won't surface if the `done` path goes through. This can be fixed by using a single channel, and checking if there was a value on receive:

```go
err, ok := <-c.err
if ok {
  // There was an error! 
  return err
}
return nil
```

But because the zero-value of the error type is `nil`, and a receive on a closed channel produces the zero-value for the type of the channel, this can be further collapsed into:
```go
return <-c.err
```

I still prefer the non-collapsed one because it shows intent better.